src/model/position.ts

```diff
@@ -57,16 +78,19 @@ export class Position {
             return 1;
         }
         const cmp = this.start.compareTo(other.start);
-        if (cmp == 0) {
+        if (cmp == 0 && this.end) {
             return this.end.compareTo(other.end);
         } else {
             return cmp;
         }
     }
 
-
+    /**
+     * If this Position has both a start and an end, and they are the same,
+     * then it is considered empty.
+     */
     isEmpty(): boolean {
-        return this.start.equals(this.end)
+        return this.end !== undefined && this.start.equals(this.end)
```

Comment:
Nella nostra position, sia start che end sono sempre valorizzati, infatti anche nel costruttore non sono nullable. Ho verificato che è così anche in Kolasu.